### PR TITLE
Travis lint for wrongly spelled PyConUK. Partly addresses Issue 36

### PR DESCRIPTION
**Pull request by snim2**

### CODE REVIEW

**Comment by snim2** [comment]
This PR should fail travis, so please don't merge yet

**Comment by snim2** [comment]
This looks better to me. Any opinons?

**Comment by ghickman** [comment]
@snim2 – This looks good to me but one thing I notice is it's not displaying where the mispellings were found. I think that will be useful output in the output of the failure case.

**Comment by inglesp** [comment]
This looks like a good approach -- but I thought we were spelling it "PyCon UK", with a space.

**Comment by snim2** [comment]
Really?

**Comment by inglesp** [comment]
Yes, according to https://github.com/PyconUK/pyconuk.github.io/commit/3475b4a19872d51a1c04ab79df439c7262e8eeea.

**Comment by snim2** [comment]
Oh hell, now it passes again :( Do you want me to squash some of these commits and force-push?

**Comment by ntoll** [comment]
PyConUK or PyconUK or PyConUk or PyCon UK or Py ConUK or... dammit... ;-) Who cares so long as we're consistent. :-D
bikeshed

**Comment by snim2** [comment]
So, the last commit gives this result on my machine (Ubuntu 15.04):

```
$ bash name-lint.sh 
index.html: <img src="static/img/header-home.jpg" alt="Best Ever Pycon UK"> index.html: <h2>Pycon UK Returns!</h2>
Please spell the conference name PyCon UK.
```

and silently exits with success on Travis. Any ideas?

**Comment by zeth** [comment]
PyCon UK has always been the name:
https://web.archive.org/web/20070427051026/http://www.pyconuk.org/

We picked it to match PyCon US and all the rest :)

**Comment by ghickman** [comment]
OK, bikes successfully shed. PyCon UK it is.

@snim2 – I've tested this locally and I'm getting the same result as Travis (no output), however I can find 83 occurrences of `Pycon UK` in the source. I've fiddled with the command and it looks like specifying the filetype globs is causing the issue.

If you change the commands to `egrep "<wrong name>" -R .` then I think we're onto a winner!

**Comment by snim2** [comment]
Thanks.

BTW the reason I didn't put this in a for-loop was that if the output of the three calls to egrep are `foo bar baz`, `monty python` and `` then BASH does this with the for loop:

    for LINT in $LINT0 $LINT1 $LINT2 ; do ... done  --> for LINT in foo bar baz monty python ; do ... done

Argh :( 

If your BASH-fu is better than mine and you know how to sort that out I'll make the change.

**Comment by snim2** [comment]
@ghickman hmmm, tried the `-R` in two places and Travis wasn't keen on either.

File: name-lint.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+LINT[0]="$(egrep -R "Pycon UK" *.html *.rst)"
+LINT[1]="$(egrep -R "pycon UK" *.html *.rst)"
+LINT[2]="$(egrep -R "pyconUK" *.html *.rst)"
```
Comment (inglesp) [inline name-lint.sh]: Could we also add "PyConUK" to the list please?

**Comment by ghickman** [comment]
@snim2 – I've had a play with this locally and come up with a more succinct version wrapping all the variations up into a single command avoiding Bashisms for any non-bash users.

``` bash
#!/bin/sh

grep -e "Pycon UK" -e "pycon UK" -e "pyconUK" -e "PyConUK" \
    --recursive . \
    --include "*.html" --include "*.rst"
if [ $? -eq 0 ]; then
    echo "Please spell the conference name PyCon UK."
    exit 1
fi
```

Hope I'm not stepping on your toes with this!

**Comment by doismellburning** [comment]
Something's very odd here:

The build of 75a6066 passes - https://travis-ci.org/PyconUK/pyconuk.github.io/builds/59835549

However, `index.html` at that commit contains the string `Pycon UK`: https://github.com/snim2/pyconuk.github.io/blob/75a60662cbef295118b0f26e729a7f35958fd4bc/index.html#L10

Running `bash name-lint.sh` locally, I get an (expected) failure:

```
$ git show -s
commit 75a60662cbef295118b0f26e729a7f35958fd4bc
Author: snim2 <mount.sarah@gmail.com>
Date:   Fri Apr 24 07:29:18 2015 +0100

    Added PyConUK to list. Use recursive grep without dereference.
$ bash name-lint.sh 
index.html:                <img src="static/img/header-home.jpg" alt="Best Ever Pycon UK">
index.html:                    <h2>Pycon UK Returns!</h2>
Please spell the conference name PyCon UK.
```

Why on earth is it passing on Travis? :s

**Comment by snim2** [comment]
@doismellburning this is why we haven't merged this PR. The lint works just fine on my machine too, so it should really fail correctly on Travis. 

I notice that Travis is running Ubuntu 12.04 (I think) and I am on 15.04. I was going to run the lint on a Ubuntu 12.04 Docker container, but I haven't had the chance yet.

**Comment by doismellburning** [comment]
I'm sorry, I keep a lot of old tabs open, and didn't refresh the page :s

**Comment by snim2** [comment]
@doismellburning if you have any other ideas please do shout. I'll have a go at some of @ghickman's ideas when I've cleared my current admin backlog :)

**Comment by snim2** [comment]
Looks like we have a winner...

File: name-lint.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+LINT="$(grep -e "Pycon UK" -e "pycon UK" -e "pyconUK" -e "PyConUK" \
+        --line-number --recursive . \
+        --include "*.html" --include "*.rst")"
+
+SUCCESS=0
+
+if [[ $LINT ]] ;  then
+    echo "${LINT[$i]}"
```
Comment (ghickman) [inline name-lint.sh]: Where does the `$i` come from here?

File: name-lint.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+LINT="$(grep -e "Pycon UK" -e "pycon UK" -e "pyconUK" -e "PyConUK" \
+        --line-number --recursive . \
+        --include "*.html" --include "*.rst")"
+
+SUCCESS=0
+
+if [[ $LINT ]] ;  then
```
Comment (ghickman) [inline name-lint.sh]: I believe you can make the same test here with single brackets making the check shell agnostic =)

Comment (snim2) [inline name-lint.sh]: @ghickman fixed I think. The latest commit works for me with `sh` and `bash` on Ubuntu 15.04 and 12.04

**Comment by ghickman** [comment]
@snim2 – This looks great, thanks for putting the time in!

**Comment by d0ugal** [comment]
Probably should have fixed the current offending cases with this :smile:

**Comment by ghickman** [comment]
@d0ugal – already on it using this handy new tool ;)

**Comment by ghickman** [comment]
Fixed names in 6a2c9bc

**Comment by doismellburning** [comment]
@ghickman Please don't do that :( Keep `master` green!

**Comment by d0ugal** [comment]
That was my thinking - new checks should come with the fixes needed ideally.

**Comment by ghickman** [comment]
I'm not personally a fan of rolling new tooling and fixes into one PR but if the consensus is to do that here then happy to do it that way =)

**Comment by ghickman** [comment]
I've just seen the readme says PRs won't get merged with failing tests so please consider me suitably chastised for my itchy merge finger!

**Comment by d0ugal** [comment]
:smile: I don't think it matters much, it is just confusing if somebody else is working on changes and sees failures between the tooling and fix PRs.

**Pull request merged**